plugins/widget/plugin.js

```diff
@@ -2917,9 +3071,10 @@
 			// Clean up and wrap all queued elements.
 			while ( ( toBeWrapped = upcastIterator.toBeWrapped.pop() ) ) {
 				cleanUpWidgetElement( toBeWrapped[ 0 ] );
-				widgetsRepo.wrapElement( toBeWrapped[ 0 ], toBeWrapped[ 1 ] );
-			}
 
+				var wrapper = widgetsRepo.wrapElement( toBeWrapped[ 0 ], toBeWrapped[ 1 ] );
+				handleStylesOnUpcast( toBeWrapped[ 0 ], wrapper );
```

Comment:
I wonder if it shouldn't be moved into upcast iterator, constructed in `createUpcastIterator`. Thanks to that we'd be able to replace some magic properties (e.g. `lockedStyles`) with reusing widget's data.

Reply:
I've moved `lockedStyle` to `data` and accessed it from here.